apps/src/lib/ui/accounts/ManageLinkedAccounts.jsx

```diff
@@ -154,26 +195,51 @@ class ManageLinkedAccounts extends React.Component {
             </tr>
           </thead>
           <tbody>
-            {this.formatAuthOptions().map(option => {
-              return (
-                <OauthConnection
-                  key={option.id || _.uniqueId('empty_')}
-                  displayName={this.getDisplayName(option.credentialType)}
-                  id={option.id}
-                  email={this.formatEmail(option)}
-                  credentialType={option.credentialType}
-                  disconnectDisabledStatus={
-                    option.id ? this.disconnectDisabledStatus(option) : null
-                  }
-                  error={option.error}
-                  personalAccountLinkingEnabled={
-                    this.props.personalAccountLinkingEnabled
-                  }
-                />
-              );
-            })}
+            {/* The options that are always available. */}
+            {this.formatAuthOptions()
```

Comment:
Since we already called `this.formatAuthOptions()` and performed a similar filter when we declared `lockedOptions`, is it possible to instead only call `this.formatAuthOptions()` once, and traverse the array only once to determine `lockedOptions` and `unlockedOptions`? Since N is super small here, it'd only be a small optimization, but would clean up this area a bit.

Reply:
It's a tricky refactor. I thought of that, too, and I guess I can build a list when I map the other.

Reply:
Yep. I'll update it to use `_.partition(...)`.